### PR TITLE
[REFACTOR] Refactor test_quantize.py to use Gluon API

### PR DESCRIPTION
**Pull request by bgawrych**
## Description ##
Refactor `tests/python/quantization/test_quantization.py` file to uutilize Gluon API instead of symbolic.
I'm enabling also flaky test for mkldnn `test_quantized_conv` as it seems to work correctly now

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage

### CODE REVIEW

**Comment by bgawrych** [comment]
@mxnet-bot run ci [centos-cpu, unix-gpu, windows-cpu]

**Review by akarbown — APPROVED** [review]

File: tests/python/quantization/test_quantization.py [hunk 1]
```diff
@@ -201,10 +195,10 @@ def check_requantize_with_symbol(shape, min_calib_range=None, max_calib_range=No
         assert_almost_equal(max_output.asnumpy(), np.array([max_output_np]))
 
     # test with symbol API.
```
Comment (bartekkuncer) [inline tests/python/quantization/test_quantization.py:197]: This comment and the one below are inconsistent. I suggest to unify them.

File: tests/python/quantization/test_quantization.py [hunk 2]
```diff
@@ -229,69 +219,106 @@ def check_quantized_conv(data_shape, kernel, num_filter, pad, stride, dilate, no
...
-        conv_exe_fp32 = conv._simple_bind(ctx=mx.current_context(), grad_req='null')
+        if len(data_shape) == 4:
+            convfp32 = mx.gluon.nn.Conv2D(channels=num_filter, kernel_size=kernel, strides=stride,
+                                          padding=pad, dilation=dilate, use_bias=use_bias)
+        elif len(data_shape) == 5:
+            convfp32 = mx.gluon.nn.Conv3D(channels=num_filter, kernel_size=kernel, strides=stride,
+                                          padding=pad, dilation=dilate, use_bias=use_bias)
+        else:
+            print('unsupported shape')
+            assert 1 == 0
```
Comment (bartekkuncer) [inline tests/python/quantization/test_quantization.py:230]: Why not just false?

File: tests/python/quantization/test_quantization.py [hunk 3]
```diff
@@ -229,69 +219,106 @@ def check_quantized_conv(data_shape, kernel, num_filter, pad, stride, dilate, no
...
-            conv_exe_int8.arg_dict[qarg_names[5]][:] = -quantized_range
-            conv_exe_int8.arg_dict[qarg_names[6]][:] = quantized_range
-            conv_exe_int8.arg_dict[qarg_names[7]][:] = -quantized_range
-            conv_exe_int8.arg_dict[qarg_names[8]][:] = quantized_range
-        qoutput, min_range, max_range = conv_exe_int8.forward()
-
-        if no_bias:
-            assert_almost_equal(output.asnumpy(), qoutput.asnumpy(), atol = 1)
-        else:
+        q_args = {
```
Comment (bartekkuncer) [inline tests/python/quantization/test_quantization.py:297]: Maybe 'qargs' to stay consistent with 'qdtype' and 'qoutput'? Or, as it is more readable with the underscore, change 'qdtype' and 'qoutput' to 'q_dtype' and 'q_output' respectively?

File: tests/python/quantization/test_quantization.py [hunk 4]
```diff
@@ -314,42 +341,46 @@ def check_quantized_elemwise_add(data_shape, qtype):
...
-        elemwise_add_int8_exe.arg_dict[qarg_names[5]][:] = data_high
-        qoutput, min_range, max_range = elemwise_add_int8_exe.forward()
+        min_dataA = mx.nd.array([data_low])
+        max_dataA = mx.nd.array([data_high])
+        min_dataB = mx.nd.array([data_low])
+        max_dataB = mx.nd.array([data_high])
+        qoutput, min_range, max_range = quantized_elemwise_add(dataA_val_int8, dataB_val_int8,
+                                                               min_dataA, max_dataA,
+                                                               min_dataB, max_dataB)
         int8_rslt = qoutput.astype(output.dtype)*max_range/0x7fffffff
```
Comment (bartekkuncer) [inline tests/python/quantization/test_quantization.py:384]: Add some spaces here please. Should not there be double '/' as division operator as we are working here on integers?

File: tests/python/quantization/test_quantization.py [hunk 5]
```diff
@@ -555,57 +612,79 @@ def maxabs(a, b):
...
+            new_args['bias'] = new_bias.astype('int32').astype('float32')
+
+        fc_fp32.load_dict(new_args, cast_dtype=True, dtype_source='saved')
+        output = fc_fp32(data)
+
+        class QuantFC(mx.gluon.nn.HybridBlock):
+            def __init__(self, num_hidden, use_bias, flatten, **kwargs):
+                super(QuantFC, self).__init__(**kwargs)
+                self.use_bias = use_bias
+                self._kwargs = {'num_hidden': num_hidden, 'no_bias': not use_bias, 'flatten': flatten}
```
Comment (bartekkuncer) [inline tests/python/quantization/test_quantization.py:624]: Maybe try aligning these variables declarations?

File: tests/python/quantization/test_quantization.py [hunk 6]
```diff
@@ -229,69 +219,106 @@ def check_quantized_conv(data_shape, kernel, num_filter, pad, stride, dilate, no
...
+            def __init__(self, channels, kernel_size, strides=(1, 1),
+                         padding=(0, 0), dilation=(1, 1), use_bias=True, **kwargs):
+                super(QuantConv, self).__init__(**kwargs)
+                self.use_bias = use_bias
+                self._kwargs = {
+                    'kernel': kernel_size, 'stride': strides, 'dilate': dilation,
+                    'pad': padding, 'num_filter': channels, 'no_bias': not use_bias, 'num_group': 1,
+                    'layout': 'NCHW'}
+
+                self.min_data = mx.gluon.Parameter('min_data', dtype='float32', allow_deferred_init=True)
```
Comment (bartekkuncer) [inline tests/python/quantization/test_quantization.py:272]: Maybe try aligning these variables declarations?

File: tests/python/quantization/test_quantization.py [hunk 7]
```diff
@@ -555,57 +612,79 @@ def maxabs(a, b):
...
+
+                if use_bias:
+                    self.bias = mx.gluon.Parameter('bias', dtype='int8', allow_deferred_init=True)
+                    self.min_bias = mx.gluon.Parameter('min_bias', dtype='float32', allow_deferred_init=True)
+                    self.max_bias = mx.gluon.Parameter('max_bias', dtype='float32', allow_deferred_init=True)
+
+            def hybrid_forward(self, F, x, weight, bias=None, min_data=None, max_data=None,
+                               min_weight=None, max_weight=None, min_bias=None, max_bias=None):
+                out = F.contrib.quantized_fully_connected(data=x, weight=weight, bias=bias, 
+                                               min_data=min_data, max_data=max_data,
```
Comment (bartekkuncer) [inline tests/python/quantization/test_quantization.py:641]: Unaligned params.

File: tests/python/quantization/test_quantization.py [hunk 8]
```diff
@@ -760,48 +864,38 @@ def check_quantized_bn(data_shape, qdtype):
...
         data = mx.nd.random.uniform(low=data_low, high=data_high, shape=data_shape)
-        gamma = mx.nd.random.uniform(low=data_low, high=data_high, shape=arg_shapes[1])
-        beta = mx.nd.random.uniform(low=data_low, high=data_high, shape=arg_shapes[2])
+        gamma = mx.nd.random.uniform(low=data_low, high=data_high, shape=fp32_params['gamma'].shape)
+        beta = mx.nd.random.uniform(low=data_low, high=data_high, shape=fp32_params['beta'].shape)
         moving_mean, moving_var = get_mean_var(data)
+        new_params = {
+            'gamma':gamma,
+            'beta':beta,
+            'running_mean': moving_mean,
```
Comment (bartekkuncer) [inline tests/python/quantization/test_quantization.py:881]: Maybe rename 'moving_*' vars to 'running_*' vars for consistency?

**Comment by bgawrych** [comment]
@mxnet-bot run ci [centos-gpu, unix-cpu, windows-cpu]

File: tests/python/quantization/test_quantization.py (hunk 4 above)
Comment (bgawrych) [reply tests/python/quantization/test_quantization.py]: Added spaces but I left equation as it was (max range seems to be float)

**Review by bartekkuncer — APPROVED** [review]

**Comment by bgawrych** [comment]
@mxnet-bot run ci [centos-gpu, unix-cpu, windows-gpu]

**Comment by bgawrych** [comment]
@mxnet-bot run ci [windows-gpu]

**Comment by bgawrych** [comment]
@mxnet-bot run ci [windows-gpu]

**Comment by bgawrych** [comment]
@mxnet-bot run ci [windows-gpu]

**Comment by bgawrych** [comment]
@mxnet-bot run ci [windows-gpu]

**Pull request merged**